### PR TITLE
Improve diagnostic configuration

### LINKED ISSUE #149: Settings: "Renpy Warn On" radio button toggles don't refresh open file(s)
Settings: "Renpy Warn On" radio button toggles (On or Off) don't take effect until manual file(s) close/load or via VSC restart 

Possible VSC technical limitation but if not it would be fantastic if the settings took immediate effect **OR** otherwise the code calls to bring up the VSC reload request info box like some theme settings currently do:
image

### PR DESCRIPTION
**Pull request by LuqueDaniel**
### Changes

- Group all diagnostic settings in `renpy.diagnostics`.
- Move diagnostics set up to `diagnostics` module.

### TODO
- [x] Add a setting to choose whether to diagnose all files in the workspace or only open files.
- [x] Fix #149

### CODE REVIEW

File: src/diagnostics.ts [hunk 1]
```diff
@@ -52,18 +52,37 @@ const rxStoreCheck = /\s+store\.(\w+)[^a-zA-Z_]?/g;
 const rxTabCheck = /^(\t+)/g;
 const rsComparisonCheck = /\s+(if|while)\s+(\w+)\s*(=)\s*(\w+)\s*/g;
 
+/**
+ * Set up diagnostics
+ * @param context extension context
+ */
+export function diagnosticsInit(context: ExtensionContext) {
+    const diagnostics = languages.createDiagnosticCollection("renpy");
```
Comment (duckdoom4) [inline src/diagnostics.ts:60]: If I recall, vscode changed the language selector API. I'd suggest using the updated API if we move this

**Review by duckdoom4 — COMMENTED** [review]
Nice cleanup! Nice work :)

File: src/diagnostics.ts (hunk 1 above)
Comment (LuqueDaniel) [reply src/diagnostics.ts:60]: Yes, If there are changes in the API we should implement them. Do you have information about those changes? I've looked at Microsoft documentation and haven't found anything.

Comment (duckdoom4) [reply src/diagnostics.ts:60]: It's this part:
https://code.visualstudio.com/api/references/vscode-api#DocumentFilter

I remember reading somewhere that they recommend using this over the single string version

Comment (duckdoom4) [reply src/diagnostics.ts:60]: But having read that again, I think in our case we don't need selectors on files that exist on disk and the default 'all' renpy files is okay.

**Comment by LuqueDaniel** [comment]
I think it's a good time to refactor the `diagnostics.ts` module and use objects.

**Review by duckdoom4 — DISMISSED** [review]

**Comment by LuqueDaniel** [comment]
@duckdoom4 Passes all checks except Format (damn pretier)

**Pull request merged**